test/jdk/java/nio/channels/unixdomain/Security.java

```diff
@@ -0,0 +1,190 @@
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Comparator;
+
+import static java.net.StandardProtocolFamily.UNIX;
+
+/**
+ * Tests required all with security manager
+ */
+
+public class Security {
+
+    static interface Command {
+        public void run() throws Exception;
+    }
+
+    static <T extends Exception> void call(Command r, Class<? extends Exception> expectedException) {
+        boolean threw = false;
+        try {
+            r.run();
+        } catch (Throwable t) {
+            if (expectedException == null) {
+                t.printStackTrace();
+                throw new RuntimeException("an exception was thrown but was not expected");
+            }
+            threw = true;
+            if (!(expectedException.isAssignableFrom(t.getClass()))) {
+                throw new RuntimeException("wrong exception type thrown " + t.toString());
+            }
+        }
+        if (expectedException != null && !threw) {
+            // should have thrown
+            throw new RuntimeException("% was expected".formatted(expectedException.getName()));
+        }
+    }
+
+
+    public static void main(String[] args) throws Exception {
+        try {
+           SocketChannel.open(UNIX);
+        } catch (UnsupportedOperationException e) {
+            System.out.println("Unix domain not supported");
+            return;
+        }
+
+        String policy = args[0];
+        switch (policy) {
+            case "policy1":
+                testPolicy1();
+                break;
+            case "policy2":
+                testPolicy2();
+                break;
+            case "policy3":
+                testPolicy3();
+                break;
+        }
+    }
+
+    static void setSecurityManager(String policy) {
+        String testSrc = System.getProperty("test.src");
+        // Three /// required for Windows below
+        String policyURL = "file:///" + testSrc + File.separator + policy;
+        System.out.println("POLICY: " + policyURL);
+        System.setProperty("java.security.policy", policyURL);
+        System.setSecurityManager(new SecurityManager());
+    }
+
+    static void close(NetworkChannel... channels) {
+
+        for (NetworkChannel chan : channels) {
+            try {
+                chan.close();
+            } catch (Exception e) {
+            }
+        }
+    }
+
+    private static final Class<SecurityException> SE = SecurityException.class;
+    private static final Class<IOException> IOE = IOException.class;
+
+    // No permission
+
+    public static void testPolicy1() throws Exception {
+        Path servername = Path.of("sock");
+        Files.deleteIfExists(servername);
+        // Permission exists to bind a ServerSocketChannel
+        final UnixDomainSocketAddress saddr = UnixDomainSocketAddress.of(servername);
+        final ServerSocketChannel server = ServerSocketChannel.open(UNIX);
+        final SocketChannel client = SocketChannel.open(UNIX);
```

Comment:
Maybe try-with-resources should be used and a finally too to ensure that the file is deleted...